view/src/extras/address.js

```diff
@@ -0,0 +1,159 @@
+import React from "react";
+import TextField from "@material-ui/core/TextField";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import LocationOnIcon from "@material-ui/icons/LocationOn";
+import Grid from "@material-ui/core/Grid";
+import Typography from "@material-ui/core/Typography";
+import { makeStyles } from "@material-ui/core/styles";
+import parse from "autosuggest-highlight/parse";
+import throttle from "lodash/throttle";
+
+function loadScript(src, position, id) {
+  if (!position) {
+    return;
+  }
+
+  const script = document.createElement("script");
+  script.setAttribute("async", "");
+  script.setAttribute("id", id);
+  script.src = src;
+  position.appendChild(script);
+}
+
+const autocompleteService = { current: null };
+
+const useStyles = makeStyles((theme) => ({
+  icon: {
+    color: theme.palette.text.secondary,
+    marginRight: theme.spacing(2),
+  },
+}));
+
+export default function GoogleMaps(props) {
+  const classes = useStyles();
+  const [value, setValue] = React.useState(props.location);
+  const [inputValue, setInputValue] = React.useState("");
+  const [options, setOptions] = React.useState([]);
+  const loaded = React.useRef(false);
+
+  if (typeof window !== "undefined" && !loaded.current) {
+    if (!document.querySelector("#google-maps")) {
+      loadScript(
+        "https://maps.googleapis.com/maps/api/js?key=AIzaSyDSPAadPOPcdIXimF1DTzKcHcp7TKa0ht8&libraries=places",
+        document.querySelector("head"),
+        "google-maps"
+      );
+    }
+
+    loaded.current = true;
+  }
+
+  const fetch = React.useMemo(
+    () =>
+      throttle((request, callback) => {
```

Comment:
I'm a little confused by this.  I'm assuming that React.useMemo() is a way of memoizing (caching) an expensive computation so you only have to do it once.  But the computation here appears to be returning a function, which is trivial.  So why not just skip the memoization and return the throttled function?

Reply:
https://reactjs.org/docs/hooks-reference.html#usememo

Reply:
You're exactly right that this is used for memoizing. The reason a function is passed to useMemo is that this function is run during rendering.